File: apis/java/src/main/java/com/github/zachdeibert/codetanks/api/model/TankInfo.java
package com.github.zachdeibert.codetanks.api.model;

public final class TankInfo implements Cloneable {
	public String name;
	public int motor;
	public int battery;
	public int cannon;
	public int radar;
	public int gps;
	public int explosives;

	@Override
	public TankInfo clone() {
		return new TankInfo(name, motor, battery, cannon, radar, gps, explosives);
	}

	public TankInfo() {
	}

	private TankInfo(String name, int motor, int battery, int cannon, int radar, int gps, int explosives) {
		this.name = name;
		this.motor = motor;
		this.battery = battery;
		this.cannon = cannon;
		this.radar = radar;
		this.gps = gps;
		this.explosives = explosives;
	}
}

File: model/TankInfo.cs
using System;
using Newtonsoft.Json;

namespace Com.GitHub.ZachDeibert.CodeTanks.Model {
    public sealed class TankInfo {
        [JsonProperty("name")]
        public string Name;
        [JsonProperty("motor")]
        public int Motors;
        [JsonProperty("battery")]
        public int Batteries;
        [JsonProperty("cannon")]
        public int Cannons;
        [JsonProperty("radar")]
        public int Radars;
        [JsonProperty("gps")]
        public int GPS;
        [JsonProperty("explosives")]
        public int Explosives;
        
        [JsonIgnore]
        public DateTime?[] CannonCharges;
        public event Action<int> CannonFire;
        public event Action Exploded;

        public void FireCannon(int i) {
            CannonFire?.Invoke(i);
        }

        public void Explode() {
            Exploded?.Invoke();
        }
    }
}
